hazelcast/src/test/java/com/hazelcast/jet/JobStatusListenerTest.java

```diff
@@ -0,0 +1,407 @@
+                    assertJobStatusEventually(job, SUSPENDED);
+                    String failure = job.getSuspensionCause().errorCause().split("\n", 3)[1];
+                    cancelAndJoin(job);
+                    assertTailEqualsEventually(listener.log,
+                            "Jet: NOT_RUNNING -> STARTING",
+                            "Jet: STARTING -> RUNNING",
+                            "Jet: RUNNING -> SUSPENDED (" + failure + ")",
+                            "User: SUSPENDED -> FAILED (CANCEL)");
+                });
+    }
+
+    @Test
+    public void testListenerDeregistration() {
+        testListener(TestSources.itemStream(1, (t, s) -> s),
+                (job, listener) -> {
+                    assertJobStatusEventually(job, RUNNING);
+                    listener.deregister();
+                    cancelAndJoin(job);
+                    assertHasNoListenerEventually(job.getIdString(), listener.registrationId);
+                    assertTailEqualsEventually(listener.log,
+                            "Jet: NOT_RUNNING -> STARTING",
+                            "Jet: STARTING -> RUNNING");
+                });
+    }
+
+    @Test
+    public void testLightListener_waitForCompletion() {
+        testLightListener(finiteStream(1, SECONDS, 2),
+                Job::join,
+                "Jet: RUNNING -> COMPLETED");
+    }
+
+    @Test
+    public void testLightListener_cancelJob() {
+        testLightListener(TestSources.itemStream(1, (t, s) -> s),
+                Job::cancel,
+                "User: RUNNING -> FAILED (CANCEL)");
+    }
+
+    @Test
+    public void testLightListener_jobFails() {
+        testLightListener(TestSources.itemStream(1, (t, s) -> 1 / (2 - s)),
+                (job, listener) -> {
+                    Throwable failure = null;
+                    try {
+                        job.getFuture().get();
+                    } catch (InterruptedException | ExecutionException e) {
+                        failure = e.getCause();
+                    }
+                    assertNotNull(failure);
+                    assertIterableEqualsEventually(listener.log,
+                            "Jet: RUNNING -> FAILED (" + failure + ")");
+                });
+    }
+
+    @Test
+    public void testLightListenerDeregistration() {
+        testLightListener(TestSources.itemStream(1, (t, s) -> s),
+                (job, listener) -> {
+                    listener.deregister();
+                    job.cancel();
+                    assertHasNoListenerEventually(job.getIdString(), listener.registrationId);
+                    assertTrue(listener.log.isEmpty());
+                });
+    }
+
+    @After
+    public void testListenerDeregistration_onCompletion() {
+        assertHasNoListenerEventually(jobIdString, registrationId);
+    }
+
+    @AfterClass
+    public static void cleanup() {
+        System.clearProperty(EventServiceImpl.EVENT_SYNC_FREQUENCY_PROP);
```

Comment:
minor: this might not be needed as the properties are restored in the serial runner.

Reply:
did you try to use `OverridePropertyRule`?